version: '3.8'

services:
  shared_models:
    build:
      context: .
      dockerfile: Dockerfile-dev
    volumes:
      - ./shared:/app/shared
      - ./validation:/app/validation
      - ./pulsar:/app/pulsar
      - ./app:/app/app
      - ./monitoring:/app/monitoring
      - ./logging:/app/logging
      - ./concurrency:/app/concurrency
      - ./main.py:/app/main.py
      - pip-cache:/home/appuser/.cache/pip
    environment:
      - PYTHONPATH=/app
      - PYTHONDONTWRITEBYTECODE=1
      - PYTHONUNBUFFERED=1
    user: "1000:1000"  # Match your host user ID for volume permissions
    command: ["python"]


volumes:
  pip-cache:
  pulsar-data: